packages/server-boilerplate/src/microService/auth/middleware.ts

```diff
@@ -8,59 +8,54 @@ import { UnauthenticatedError } from '@tupaia/utils';
 import { AccessPolicy } from '@tupaia/access-policy';
 import {
   Authenticator,
-  encryptPassword,
-  getUserIDFromToken,
+  getTokenClaimsFromBearerAuth,
   getUserAndPassFromBasicAuth,
 } from '@tupaia/auth';
 import { ModelRegistry } from '@tupaia/database';
 
-const authenticateBearerAuthHeader = async (authHeader: string) => {
+import { AccessPolicyObject } from '../../types';
+
+const getBearerAccessPolicy = async (
+  authenticator: Authenticator,
+  authHeader: string,
+): Promise<AccessPolicyObject> => {
   // Use the user account provided in the auth header if present
-  const tokenUserID = getUserIDFromToken(authHeader);
-  if (tokenUserID) {
-    return tokenUserID;
+  const { userId, apiClientUserId } = getTokenClaimsFromBearerAuth(authHeader);
+  if (!userId) {
+    throw new UnauthenticatedError('Could not authenticate with the provided access token');
   }
 
-  throw new UnauthenticatedError('Could not authenticate with the provided access token');
-};
-
-const attemptApiClientAuthentication = async (
-  models: ModelRegistry,
-  { username, secretKey }: { username: string; secretKey: string },
-) => {
-  const secretKeyHash = encryptPassword(secretKey, process.env.API_CLIENT_SALT);
-  const apiClient = await models.apiClient.findOne({
-    username,
-    secret_key_hash: secretKeyHash,
-  });
-  return apiClient && apiClient.getUser();
+  const { accessPolicy } = await authenticator.authenticateAccessToken({ userId, apiClientUserId });
+  return accessPolicy;
 };
 
-const authenticateBasicAuthHeader = async (
+const getBasicAccessPolicy = async (
   models: ModelRegistry,
   authenticator: Authenticator,
   authHeader: string,
   apiName: string,
-) => {
+): Promise<AccessPolicyObject> => {
   const { username, password } = getUserAndPassFromBasicAuth(authHeader);
 
   // first attempt to authenticate as an api client, in case a secret key was used in the auth header
-  const apiClientUser = await attemptApiClientAuthentication(models, {
-    username,
-    secretKey: password,
-  });
-  if (apiClientUser) {
-    return apiClientUser.id;
+  try {
+    const { accessPolicy } = await authenticator.authenticateApiClient({
+      username,
+      secretKey: password,
+    });
+    return accessPolicy;
+  } catch (error) {
+    // attempting to authenticate as api client failed
```

Comment:
Silently failing here as if the user isn't an `apiClient` then this call will throw and error (that keeps it consistent with the other functions on `Authenticator`)